src/Accounts/Authentication/Config/ConfigInitializer.cs

```diff
@@ -196,6 +196,13 @@ private void RegisterConfigs(IConfigManager configManager)
                 true,
                 null,
                 new[] { AppliesTo.Az }));
+            //Use DisableErrorRecordsPersistence as opt-out for now, will replace it with EnableErrorRecordsPersistence as opt-in at next major release (November 2023)
+            configManager.RegisterConfig(new SimpleTypedConfig<bool>(
+                ConfigKeys.DisableErrorRecordsPersistence,
+                Resources.HelpMessageOfDisableErrorRecordsPersistence,
+                false,
+                ConfigKeys.DisableErrorRecordsPersistence,
```

Comment:
As a user, I'd prefer the environment variable to have some prefix to indicate that it only applies to Azure PowerShell, i.e. `AzPSDisable...`